reconcile/statuspage/integrations/maintenances.py

```diff
@@ -80,6 +80,7 @@ def run(self, dry_run: bool = False) -> None:
                     for m in p.maintenances or []
                     if datetime.fromisoformat(m.scheduled_start) > now
                 ]
+                desired_state = sorted(desired_state, key=lambda d: d.name)
```

Comment:
is sorting required in some case?

Reply:
`name` need to be unique, otherwise diff result may miss things (last one wins)

https://github.com/app-sre/qontract-reconcile/blob/60c897ef52b21ae05ec84096e431489fd559a90b/reconcile/statuspage/integrations/maintenances.py#L34

Reply:
> is sorting required in some case?

yes, I saw the case while debugging. desired and current state had reverse orders

> name need to be unique

yes, shoudl be good as per schema: https://github.com/app-sre/qontract-schemas/blob/main/graphql-schemas/schema.yml#L4358

Reply:
if it's unique, then the order doesn't matter, as they will be converted to dict by key during diff